languages/tree-sitter-stack-graphs-python/src/stack-graphs.tsg

```diff
@@ -794,19 +794,29 @@ inherit .parent_module
   edge @name.def -> @func.call
 
   ; Prevent functions defined inside of method bodies from being treated like methods
+  ; There might be a bug here as calling .self_scope after this, will not fallback to the larger scope
```

Comment:
Do you have an idea what program might trigger this? Even if we don't fix it now, we can document it by adding test and giving it the `.skip` suffix.

Reply:
It does not enters this branch anymore now that we only target method definitions (first level functions in classes). But I started testing stuff on [tree-sitter-graph](https://github.com/tree-sitter/tree-sitter-graph/) and the following stanzas reproduce this behaviour:

```tsg
inherit .scope
(module) @mod {
    let @mod.scope = "module"
}

(function_definition) @def {
    let @def.scope = #null
}

(pass_statement) @pass {
    print @pass.scope ; will be null, not module
}
```

```
def f():
    pass
```

I don't think it is an issue anymore in stack-graphs tho (at least for python).
Maybe this is the expected behaviour but then being able to set a property to "undefined" / unset it would be useful. Let's track this in [tree-sitter-graph](https://github.com/tree-sitter/tree-sitter-graph/).

Reply:
I'm not sure I completely understand the issue, but if this cannot be triggered by a Python program here, let's remove the comment.

Reply:
I'll make sure to open an issue on tree sitter graph, just to keep track of it (as it might be an issue in the future), I removed the comment.
I'm not sure those two lines are needed at all now, but I might have missed something (and it's out of this pr scope I believe)
```
  ; Prevent functions defined inside of method bodies from being treated like methods
  let @body.class_self_scope = #null
  let @body.class_member_attr_scope = #null
 ```